src/Stratis.Bitcoin.IntegrationTests.Common/TestHelper.cs

```diff
@@ -11,17 +12,23 @@ namespace Stratis.Bitcoin.IntegrationTests.Common
 {
     public class TestHelper
     {
-        public static void WaitLoop(Func<bool> act, string failureReason = "Unknown Reason", int millisecondsTimeout = 50, CancellationToken cancellationToken = default(CancellationToken))
+        static async Task AsyncTaskDelay(int milisec)
+        {
+            await Task.Delay(milisec).ConfigureAwait(false);
+        }
+
+        public static void WaitLoop(Func<bool> act, string failureReason = "Unknown Reason", int millisecondsTimeout = 150, CancellationToken cancellationToken = default(CancellationToken))
         {
             cancellationToken = cancellationToken == default(CancellationToken)
-                ? new CancellationTokenSource(Debugger.IsAttached ? 15 * 60 * 1000 : 30 * 1000).Token
+                ? new CancellationTokenSource(Debugger.IsAttached ? 15 * 60 * 1000 : 40 * 1000).Token
                 : cancellationToken;
+
             while (!act())
             {
                 try
                 {
                     cancellationToken.ThrowIfCancellationRequested();
-                    Thread.Sleep(millisecondsTimeout);
+                    AsyncTaskDelay(millisecondsTimeout).Wait();
```

Comment:
So I think the `Wait()` here defeats the purpose of having a `Task.Delay` as the task is waited on. 
Running the code, I saw no difference in speed or reliability.

Reply:
https://stackoverflow.com/questions/34052381/thread-sleep2500-vs-task-delay2500-wait

Reply:
accept